tests/phpunit/tests/rest-api/rest-global-styles-controller.php

```diff
@@ -116,17 +212,61 @@ public function test_update_item() {
 		$this->assertEquals( 'My new global styles title', $data['title']['raw'] );
 	}
 
+
+	/**
+	 * @covers WP_REST_Global_Styles_Controller::update_item
+	 * @ticket 54516
+	 */
+	public function test_update_item_no_user() {
+		wp_set_current_user( 0 );
+		$request  = new WP_REST_Request( 'PUT', '/wp/v2/global-styles/' . self::$global_styles_id );
+		$response = rest_get_server()->dispatch( $request );
+		$this->assertErrorResponse( 'rest_cannot_edit', $response, 401 );
+	}
+
+	/**
+	 * @covers WP_REST_Global_Styles_Controller::update_item
+	 * @ticket 54516
+	 */
+	public function test_update_item_invalid_post() {
+		wp_set_current_user( self::$admin_id );
+		$request  = new WP_REST_Request( 'PUT', '/wp/v2/global-styles/' . self::$post_id );
+		$response = rest_get_server()->dispatch( $request );
+		$this->assertErrorResponse( 'rest_global_styles_not_found', $response, 404 );
+	}
+
+	/**
+	 * @covers WP_REST_Global_Styles_Controller::update_item
+	 * @ticket 54516
+	 */
+	public function test_update_item_permission_check() {
+		wp_set_current_user( self::$subscriber_id );
+		$request  = new WP_REST_Request( 'PUT', '/wp/v2/global-styles/' . self::$global_styles_id );
+		$response = rest_get_server()->dispatch( $request );
+		$this->assertErrorResponse( 'rest_cannot_edit', $response, 403 );
+	}
+
 	public function test_delete_item() {
-		$this->markTestIncomplete();
+		$this->markTestSkipped( 'Controller does not implement delete_item().' );
 	}
 
 	public function test_prepare_item() {
-		// TODO: Implement test_prepare_item() method.
-		$this->markTestIncomplete();
+		$this->markTestSkipped( 'Controller does not implement prepare_item().' );
 	}
 
+	/**
+	 * @covers WP_REST_Global_Styles_Controller::get_item_schema
+	 * @ticket 54516
+	 */
 	public function test_get_item_schema() {
-		// TODO: Implement test_get_item_schema() method.
-		$this->markTestIncomplete();
+		$request    = new WP_REST_Request( 'OPTIONS', '/wp/v2/global-styles/' . self::$global_styles_id );
+		$response   = rest_get_server()->dispatch( $request );
+		$data       = $response->get_data();
+		$properties = $data['schema']['properties'];
+		$this->assertCount( 4, $properties );
+		$this->assertArrayHasKey( 'id', $properties );
+		$this->assertArrayHasKey( 'styles', $properties );
+		$this->assertArrayHasKey( 'settings', $properties );
+		$this->assertArrayHasKey( 'title', $properties );
```

Comment:
Add error messages to identify which assertion fails (if any fail) as the others after it will not run. This helps with debugging failed tests.

```suggestion 
		$this->assertCount( 4, $properties, 'Schema properties array does not have exactly 4 elements' );
		$this->assertArrayHasKey( 'id', $properties, 'Schema properties array does not have "id" key'  );
		$this->assertArrayHasKey( 'styles', $properties, 'Schema properties array does not have "styles" key'  );
		$this->assertArrayHasKey( 'settings', $properties, 'Schema properties array does not have "settings" key'  );
		$this->assertArrayHasKey( 'title', $properties, 'Schema properties array does not have "title" key'  );
```